server/src/logic.rs

```diff
@@ -79,6 +126,22 @@ async fn handle_enumerate(
         uri
     );
 
+    let collector_hostname = if let Some(hostname) = collector.hostname() {
+        hostname.to_owned()
+    } else {
+        let hostname = message.header().to().and_then(|to| {
+            let url = url::Url::parse(to).ok()?;
+            url.host_str().map(|s| s.to_owned())
+        });
+
+        let Some(hostname) = hostname else {
+            error!("Collector's hostname cannot be extracted");
```

Comment:
Could you add the value of `message.header().to()` to the log? This will be useful to understand what happened.

Also, I would use the "warn" log level here since it's not an openwec error but rather an inconsistent input value from the client.

Reply:
Do you mean `message.header()`? `message.header().to()` is `None` in this case.

Reply:
I meant `message.header().to()`. You can display it using debug format `{:?}`. Correct me if I'm wrong:  if it is `None`, it means that the header was not present in the client request, which is interesting to know. Otherwise, it means that there is something wrong with the url the the client provided.

Another approach would be to split this log message into two: "Missing 'To' header in client request" and "Invalid 'To' header in client request: {}".

Reply:
Now I get it, thanks.

Reply:
Looks better:
```
WARN server::logic - Collector hostname cannot be extracted from To header (Some("I'm not an URL")), rejecting Enumerate request
```